### PR TITLE
Make metrics of raftstore worker static & thread localize

### PR DESCRIPTION
**Pull request by Renkai**
### What problem does this PR solve?

Issue Number: Part of https://github.com/tikv/tikv/issues/5708

Problem Summary:

### What is changed and how it works?

Try to make metrics more effecient

### Related changes

- No

### Check List 

Tests 

- Unit test

### Release note

### CODE REVIEW

**Comment by Renkai** [comment]
@sre-bot /run-all-tests

**Comment by Renkai** [comment]
@sre-bot /release make dist_release

**Comment by Renkai** [comment]
@sre-bot /bench +sysbench +tpcc +tpch

File: components/raftstore/src/store/worker/region.rs [hunk 1]
```diff
@@ -263,11 +263,8 @@ impl<R: CasualRouter<RocksEngine>> SnapContext<R> {
         kv_snap: RocksSnapshot,
         notifier: SyncSender<RaftSnapshot>,
     ) {
-        SNAP_COUNTER_VEC
-            .with_label_values(&["generate", "all"])
-            .inc();
-        let gen_histogram = SNAP_HISTOGRAM.with_label_values(&["generate"]);
-        let timer = gen_histogram.start_coarse_timer();
+        SNAP_COUNTER.generate.all.inc();
+        let begin_instant = tikv_util::time::Instant::now();
```
Comment (zhangjinpeng1987) [inline components/raftstore/src/store/worker/region.rs]: s/begin_instant/start

File: components/raftstore/src/store/worker/region.rs [hunk 2]
```diff
@@ -368,35 +365,32 @@ impl<R: CasualRouter<RocksEngine>> SnapContext<R> {
     /// Tries to apply the snapshot of the specified Region. It calls `apply_snap` to do the actual work.
     fn handle_apply(&mut self, region_id: u64, status: Arc<AtomicUsize>) {
         status.compare_and_swap(JOB_STATUS_PENDING, JOB_STATUS_RUNNING, Ordering::SeqCst);
-        SNAP_COUNTER_VEC.with_label_values(&["apply", "all"]).inc();
-        let apply_histogram = SNAP_HISTOGRAM.with_label_values(&["apply"]);
-        let timer = apply_histogram.start_coarse_timer();
+        SNAP_COUNTER.apply.all.inc();
+        // let apply_histogram = SNAP_HISTOGRAM.with_label_values(&["apply"]);
+        // let timer = apply_histogram.start_coarse_timer();
+        let begin_instant = tikv_util::time::Instant::now();
```
Comment (zhangjinpeng1987) [inline components/raftstore/src/store/worker/region.rs]: ditto

**Pull request merged**